### PR TITLE
[fix](nereids)add visitMarkJoinReference method in ExpressionDeepCopier

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java [hunk 1]
```diff
@@ -75,6 +76,20 @@ public Expression visitSlotReference(SlotReference slotReference, DeepCopierCont
         }
     }
 
+    @Override
+    public Expression visitMarkJoinReference(MarkJoinSlotReference slotReference, DeepCopierContext context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java:80]: check all namedExpression please~

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java [hunk 2]
```diff
@@ -75,6 +79,62 @@ public Expression visitSlotReference(SlotReference slotReference, DeepCopierCont
...
+            return slotReference.withExprId(newExprId);
+        } else {
+            MarkJoinSlotReference newOne = slotReference.withExprId(StatementScopeIdGenerator.newExprId());
+            exprIdReplaceMap.put(slotReference.getExprId(), newOne.getExprId());
+            return newOne;
+        }
+    }
+
+    @Override
+    public Expression visitArrayItemSlot(ArrayItemReference.ArrayItemSlot slotReference, DeepCopierContext context) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java:96]: use withExprId, and we do not need visitXXXSlot method.

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java [hunk 3]
```diff
@@ -276,7 +276,7 @@ public Expr visitSlotReference(SlotReference slotReference, PlanTranslatorContex
     }
 
     @Override
-    public Expr visitArrayItemSlot(SlotReference slotReference, PlanTranslatorContext context) {
+    public Expr visitArrayItemSlot(ArrayItemReference.ArrayItemSlot slotReference, PlanTranslatorContext context) {
         return context.findColumnRef(slotReference.getExprId());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java:280]:
```suggestion
    public Expr visitArrayItemSlot(ArrayItemReference.ArrayItemSlot arrayItemSlot, PlanTranslatorContext context) {
        return context.findColumnRef(arrayItemSlot.getExprId());
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java [hunk 4]
```diff
@@ -63,13 +65,27 @@ public Expression visitAlias(Alias alias, DeepCopierContext context) {
...
+            newExprId = exprIdReplaceMap.get(slotReference.getExprId());
+        } else {
+            newExprId = StatementScopeIdGenerator.newExprId();
+            exprIdReplaceMap.put(slotReference.getExprId(), newExprId);
+        }
+        return slotReference.withExprId(newExprId);
+    }
+
+    @Override
+    public Expression visitArrayItemReference(ArrayItemReference slotReference, DeepCopierContext context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java:80]:
```suggestion
    public Expression visitArrayItemReference(ArrayItemReference arrayItemReference, DeepCopierContext context) {
```

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java [hunk 5]
```diff
@@ -64,13 +66,27 @@ public Expression visitAlias(Alias alias, DeepCopierContext context) {
...
+        return slotReference.withExprId(newExprId);
+    }
+
+    @Override
+    public Expression visitArrayItemReference(ArrayItemReference arrayItemSlot, DeepCopierContext context) {
+        Map<ExprId, ExprId> exprIdReplaceMap = context.exprIdReplaceMap;
+        if (exprIdReplaceMap.containsKey(arrayItemSlot.getExprId())) {
+            ExprId newExprId = exprIdReplaceMap.get(arrayItemSlot.getExprId());
+            return new ArrayItemReference(newExprId, arrayItemSlot.getName(),
+                    arrayItemSlot.getArrayExpression());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java:85]: do we need rewrite the expression returned by arrayItemSlot.getArrayExpression()?

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**